LK-MPI documentation update (#1190)
PennyLaneAI/pennylane-lightning: doc/lightning_gpu/device.rst

        return qml.state()
    local_state_vector = circuit_mpi()

> .. note::

Reviewer: To fix the rendering issue on Brave/Win:

```suggestion
.. note::

```

>     The total number of MPI processes and MPI processes per node must be powers of 2. For example, 2, 4, 8, 16, etc. Each MPI process is responsible for managing one GPU. 

Reviewer: is this "for example" really needed? :D 
```suggestion
    The total number of MPI processes must be powers of 2, where each MPI process is responsible for managing one GPU. 
```

Currently, a ``lightning.gpu`` device with the MPI multi-GPU backend supports all the ``gate operations`` and ``observables`` that a ``lightning.gpu`` device with a single GPU/node backend supports.

By default, each MPI process will return the overall simulation results, except for the ``qml.state()`` and ``qml.probs()`` methods for which each MPI process only returns the local simulation

> results for the ``qml.state()`` and ``qml.prob()`` methods to avoid buffer overflow. It is the user's responsibility to ensure correct data collection for those two methods. Here are examples of collecting

Reviewer: 🫠 
```suggestion
results for the ``qml.state()`` and ``qml.probs()`` methods to avoid buffer overflow. It is the user's responsibility to ensure correct data collection for those two methods. Here are examples of collecting
```
